concrete/src/Job/Service.php

```diff
@@ -0,0 +1,117 @@
+<?php
+namespace Concrete\Core\Job;
+
+use Concrete\Core\Application\Application;
+use Concrete\Core\Database\Connection\Connection;
+
+class Service
+{
+    protected $app;
+
+    public function __construct(Application $app)
+    {
+        $this->app = $app;
+    }
+
+    /**
+     * @param string $jHandle
+     * @param object $pkg
+     *
+     * @return Job|false
+     */
+    public function installByPackage($jHandle, $pkg)
```

Comment:
Didn't see your response to my comment before, my recommendation was 2 fold:
1. This service class should not be creating jobs, that should be happening in the JobFactory no?
2. I don't think ->installByPackage and ->installByHandle need to be here. I think the pattern could instead be:

```php
$jobService->install($job_object_or_handle, $pkg);
```
and the install method could make use of the JobFactory class to get what it needs. Ideally JobFactory would be a hard dependency of this service meaning it'd be a required parameter on this classes constructor.